### PR TITLE
+ Add disabled prop to SingleSelect component

### PR DESCRIPTION
**Pull request by gdostie**

### CODE REVIEW

File: src/components/select/SingleSelectConnected.tsx [hunk 1]
```diff
@@ -65,6 +67,7 @@ export class SingleSelectConnected extends React.Component<ISingleSelectProps, {
                 onMouseUp={props.onMouseUp}
                 onKeyDown={props.onKeyDown}
                 onKeyUp={props.onKeyUp}
+                disabled={this.props.disabled}
```
Comment (pochretien) [inline src/components/select/SingleSelectConnected.tsx]: {...this.props.buttonProps}

File: src/components/select/SingleSelectConnected.tsx [hunk 2]
```diff
@@ -13,6 +13,7 @@ export interface ISingleSelectOwnProps extends ISelectProps {
     placeholder?: string;
     toggleClasses?: string;
     onSelectOptionCallback?: (option: string) => void;
+    disabled?: boolean;
```
Comment (pochretien) [inline src/components/select/SingleSelectConnected.tsx]: buttonProps: React.HTMLProps<HTMLButtonElement>

Comment (GermainBergeron) [reply src/components/select/SingleSelectConnected.tsx]: I would even go further and remove it. We should go the same way as #644 by using the `React.Component<ISingleSelectProps & React.HTMLProps<HTMLButtonElement>` in combinaison with `ts-transformer-keys` and the `{..._.omit(this.props, keys<ISingleSelectProps>()}`

**Comment by gdostie** [comment]
demo: https://coveo.github.io/react-vapor/add-singleselect-disabled-prop/

**Pull request merged**